### PR TITLE
K8SPSMDB-319: Add connection endpoint to cr output

### PR DESCRIPTION
**Pull request by qJkee**
[K8SPSMDB-319](https://jira.percona.com/browse/K8SPSMDB-319)

Show Endpoint in psmbd custom resource output

To test with replica set
`kubectl exec psmdb-client-bbfddc864-44v9l -- bash -c 'printf '\''db.runCommand({connectionStatus:1,showPrivileges:true})\n'\'' | mongo mongodb+srv://clusterMonitor:clusterMonitor123456@<ENDPOINT OUTPUT>/admin?ssl=false\&replicaSet=rs0 '`

To test with sharding
`kubectl exec psmdb-client-bbfddc864-44v9l -- bash -c 'printf '\''db.runCommand({ping : 1})\n'\'' | mongo mongodb://clusterMonitor:clusterMonitor123456@<ENDPOINT OUTPUT>/admin?ssl=false'`

### CODE REVIEW

File: pkg/controller/perconaservermongodb/status.go [hunk 1]
```diff
@@ -233,3 +239,39 @@ func (r *ReconcilePerconaServerMongoDB) rsStatus(rsSpec *api.ReplsetSpec, cluste
...
+		if mongos := cr.Spec.Sharding.Mongos; mongos.Expose.Enabled &&
+			mongos.Expose.ExposeType == corev1.ServiceTypeLoadBalancer {
+			return loadBalancerServiceEndpoint(r.client, cr.Name+"-mongos", cr.Namespace)
+		}
+		return cr.Name + "-mongos." + cr.Namespace + "." + cr.Spec.ClusterServiceDNSSuffix, nil
+	}
+
+	if rs := cr.Spec.Replsets[0]; rs.Expose.Enabled &&
+		rs.Expose.ExposeType == corev1.ServiceTypeLoadBalancer {
+		return loadBalancerServiceEndpoint(r.client, cr.Name+"-"+cr.Spec.Replsets[0].Name, cr.Namespace)
```
Comment (delgod) [inline pkg/controller/perconaservermongodb/status.go:254]: in such case, I think we should return a comma-separated list of all load balancers.

like this
```
     addrs, err := psmdb.GetReplsetAddrs(c, cluster, rs, pods.Items)
     if err != nil {
         return nil, errors.Wrap(err, "get mongo addr")
     }
     return strings.Join(addrs, ","), nil
```

**Review by delgod — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-475-546090cf

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade | passed |
| self-healing | passed |
| limits | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| scaling | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| security-context | passed |
| operator-self-healing | passed |
| arbiter | passed |
| service-per-pod | passed |
| scheduled-backup | passed |
| liveness | passed |
| demand-backup-sharded | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

File: pkg/controller/perconaservermongodb/status.go (hunk 1 above)
Comment (qJkee) [reply pkg/controller/perconaservermongodb/status.go]: ca6fa9c

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-475-ca6fa9c2

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| arbiter | passed |
| service-per-pod | passed |
| scheduled-backup | passed |
| liveness | passed |
| demand-backup-sharded | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-475-44fd5823

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| upgrade | passed |
| storage | passed |
| monitoring | passed |
| self-healing | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| demand-backup-sharded | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |

**Review by hors — APPROVED** [review]

**Review by nonemax — APPROVED** [review]

**Pull request merged**